src/gql/GQLWrapper.tsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import { ApolloClient } from "apollo-client";
+import { ApolloProvider } from "@apollo/react-hooks";
+import { GraphQLSchema } from "graphql/type";
+import { HttpLink } from "apollo-link-http";
+import { InMemoryCache, NormalizedCacheObject } from "apollo-cache-inmemory";
+import { introspectSchema, makeExecutableSchema } from "graphql-tools";
+import { printSchema } from "graphql/utilities/schemaPrinter";
+import { SchemaLink } from "apollo-link-schema";
+import { useEffect, useState } from "react";
+import resolvers from "./resolvers";
+
+interface ClientLinkParams {
+  gqlURL?: string;
+  isDevelopment?: boolean;
+  isTest?: boolean;
+  schemaString?: string;
+}
+
+interface Props extends ClientLinkParams {
+  children?: React.ReactNode;
+}
+
+export async function getClientLink({
+  gqlURL,
+  isDevelopment,
+  isTest,
+  schemaString
+}: ClientLinkParams): Promise<HttpLink | SchemaLink> {
+  const httpLink: HttpLink = new HttpLink({
+    uri: gqlURL
+  });
+
+  if (isDevelopment || isTest) {
+    let executableSchema: GraphQLSchema | null = null;
+    try {
+      executableSchema = makeExecutableSchema({
+        typeDefs: schemaString
+          ? schemaString
+          : printSchema(await introspectSchema(httpLink)),
+        resolvers
+      });
+    } catch (e) {
+      console.log(
```

Comment:
@tgrander This catch block can only be entered during dev mode and the message is intended for the developer only. The message means that one of two variables should be set if the developer wants to dev against a mock server and serves as guidance if the developer is not seeing what they expect. 
It may be possible that this is what the developer intended because they wanted to see the outcome of failed requests or view the user experience when the App cannot connect to GQL. 
Does this change your mind about logging to the console? Otherwise I can remove the message.

Reply:
i am still suspicious of putting console.logs in production even if in just dev mode. but if that's what you'd like to do then i won't stand in your way. can you change it to `console.error` though? also would `throw new Error()`be helpful to the dev or no? for example in CRA if there is an error in dev then it shows red screen of death that makes error very clear to developer which helps to resolve. this error will fail silently without dev knowing exactly what happened without opening console.